wcfsetup/install/files/lib/action/RegisterServiceWorkerAction.class.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace wcf\action;
+
+use Laminas\Diactoros\Response\EmptyResponse;
+use Laminas\Diactoros\Response\TextResponse;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use wcf\data\service\worker\ServiceWorkerEditor;
+use wcf\data\service\worker\ServiceWorkerList;
+use wcf\http\Helper;
+use wcf\system\exception\PermissionDeniedException;
+use wcf\system\WCF;
+
+/**
+ * @author      Olaf Braun
+ * @copyright   2001-2024 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.1
+ */
+final class RegisterServiceWorkerAction implements RequestHandlerInterface
+{
+    #[\Override]
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        if ($request->getMethod() !== 'POST') {
+            return new TextResponse('Unsupported', 400);
+        }
+        if (!WCF::getUser()->userID) {
+            // Notification not sent to guests
+            throw new PermissionDeniedException();
+        }
+        $parameters = Helper::mapRequestBody(
+            $request->getParsedBody(),
+            <<<'EOT'
+                array {
+                    remove: bool,
+                    endpoint: non-empty-string,
+                    publicKey: non-empty-string,
+                    authToken: non-empty-string,
+                    contentEncoding: "aesgcm" | "aes128gcm",
+                }
+            EOT,
```

Comment:
```suggestion
                EOT,
```

Indentation matters here. Currently all lines are intended by four additional spaces, because PHP only trims whitespace that aligns with the identation of the ending marker.